apps/meteor/app/reactions/server/setReaction.js

```diff
@@ -84,6 +107,13 @@ async function setReaction(room, user, message, reaction, shouldReact) {
 			};
 		}
 		message.reactions[reaction].usernames.push(user.username);
+		if (settings.get('UI_Use_Real_Name') && !message.reactions[reaction].names) {
+			message.reactions[reaction].names = message.reactions[reaction].usernames.map((username) =>
+				getNameByUsername(username, message, reaction),
+			);
+		} else if (message.reactions[reaction].names) {
+			message.reactions[reaction].names.push(user.name);
+		}
```

Comment:
Conditionally pushing is a bad idea here. You're inserting incomplete data, further creating the need to modify it on runtime. Why is username a must field but names is conditional?

Reply:
I thought `names` may not be an expected field when `UI_Use_Real_Name` is false but we can make it a must field as well if you say.

Reply:
Also names array is optional in the current [type](https://github.com/RocketChat/Rocket.Chat/blob/bf915717dbb1d0f4ba46df71f682fbb815ddcbe2/packages/core-typings/src/IMessage/IMessage.ts#L166).

Reply:
> I thought names may not be an expected field when UI_Use_Real_Name is false
We should still add that or else we incur the cost of having to resolve those once the setting is updated.